pkg/datastore/kubernetes/kubernetes.go

```diff
@@ -211,10 +254,27 @@ func (k *Datastore) WatchEndpoints(ctx context.Context, selfClusterID string, co
 				klog.V(log.DEBUG).Infof("Recovered deleted object '%s' from tombstone", object.GetName())
 			}
 
-			if selfClusterID != object.Spec.ClusterID {
-				utilruntime.HandleError(onEndpointChange(&types.SubmarinerEndpoint{
-					Spec: object.Spec,
-				}, false))
+			clusters, err := k.GetClusters(colorCodes)
+			if err != nil {
+				utilruntime.HandleError(err)
+			}
+
+			klog.V(log.DEBUG).Infof("Got clusters : %#v", clusters)
+			for _, cluster := range clusters {
+				endpoints, err := k.GetEndpoints(cluster.Spec.ClusterID)
+				if err != nil {
+					utilruntime.HandleError(err)
+					continue
+				}
+
+				klog.V(log.DEBUG).Infof("Got endpoints : %#v", endpoints)
+				for _, endpoint := range endpoints {
+					if selfClusterID != endpoint.Spec.ClusterID {
+						utilruntime.HandleError(onEndpointChange(&types.SubmarinerEndpoint{
+							Spec: endpoint.Spec,
+						}, false))
+					}
+				}
```

Comment:
Why do we notify `onEndpointChange` of `Endpoints` from all matching clusters instead of just the one that was added?